instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/instrumentation.rb

```diff
@@ -49,17 +61,14 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         option :enable_platform_authorized,   default: false, validate: :boolean
         option :enable_platform_resolve_type, default: false, validate: :boolean
         option :legacy_platform_span_names,   default: false, validate: :boolean
+        option :legacy_tracing,               default: false, validate: :boolean
```

Comment:
GraphQL 2 does indeed support the older API. I think we have the option in place here to ensure we can disable it in production workloads in case there is a bug or perf issue we missed. 

@rmosolgo do you have any timelines for when the 1.0 gem will no longer be actively maintained and when the older trace interface will be deprecated?

Reply:
I'd expect to remove the old tracing API in GraphQL-Ruby 3.0, with backwards-compatibility until then. (In GraphQL-Ruby, backwards compatibility is zero-overhead when it's not used) I don't have plans for GraphQL-Ruby 3.0 in the near (or distant) future.